litejpeg/core/rle/rlecore.py

```diff
@@ -0,0 +1,316 @@
+"""
+RLE Core Module:
+----------------
+This module is the core module for the RLE for calculating
+the amplitude and the zero_count of the input data.
+amplitudes are the non-zero elements present in the matrix
+and zero_count is the number of zeros before the non-zero
+amplitude.
+This helps is the reduction of data as no more memory is
+required to store all the zeros instead a runlength is
+sufficient for storing all the zeros.
+"""
+
+# Importing libraries.
+from litex.gen import *
+from litex.soc.interconnect.stream import *
+
+from litejpeg.core.common import *
+
+# To provide delay so in order to sink the data coming from the main
+# module to that of the Datapath module.
+datapath_latency = 3
+
+
+# CEInseter genenrates an additional instance named as self.ce
+# This is used so that to provide an additional clock attached to the
+# pipeline Actor in the main module.
+@CEInserter()
+
+
+class RLEDatapath(Module):
+    """
+    RLEDatapath Module:
+    -------------------
+    This module is the datapath set for the steps required for
+    the RLE module to take place.
+
+    The input matrix contains two components,
+    the first value of the matrix is called as the DC component and
+    the rest of the values within the matrix are the AC component.
```

Comment:
This is much better description of the input. It explains what the parts are.
```
    The input matrix contains two components,
     * the first value of the matrix is the DC component, and
     * the rest of the values within the matrix are the AC component.
```

Reply:
DONE